pkg/apis/numaflow/v1alpha1/source.go

```diff
@@ -108,3 +113,39 @@ func (s Source) getUDTransformerContainer(mainContainerReq getContainerReq) core
 	}
 	return container
 }
+
+func (s Source) getUDSourceContainer(mainContainerReq getContainerReq) corev1.Container {
+	c := containerBuilder{}.
+		name(CtrUdsource).
+		imagePullPolicy(mainContainerReq.imagePullPolicy). // Use the same image pull policy as the main container
+		appendVolumeMounts(mainContainerReq.volumeMounts...)
+	if x := s.UDSource.Container; x != nil && x.Image != "" { // customized image
+		c = c.image(x.Image)
+		if len(x.Command) > 0 {
+			c = c.command(x.Command...)
+		}
+		if len(x.Args) > 0 {
+			c = c.args(x.Args...)
+		}
+	}
+	if x := s.UDSource.Container; x != nil {
+		c = c.appendEnv(x.Env...).appendVolumeMounts(x.VolumeMounts...).resources(x.Resources).securityContext(x.SecurityContext).appendEnvFrom(x.EnvFrom...)
+		if x.ImagePullPolicy != nil {
+			c = c.imagePullPolicy(*x.ImagePullPolicy)
+		}
+	}
+	container := c.build()
+	container.LivenessProbe = &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "/sidecar-livez",
```

Comment:
Do we have the logic change in `/sidecar-livez`?

Reply:
synced offline, I will address this comment in follow-up PRs.